File: Leetcode/Q66.py
"""
You are given a large integer represented as an integer array digits, where each digits[i] is the ith digit of the integer. The digits are ordered from most significant to least significant in left-to-right order. The large integer does not contain any leading 0's.

Increment the large integer by one and return the resulting array of digits.


Example 1:
Input: digits = [1,2,3]
Output: [1,2,4]
Explanation: The array represents the integer 123.
Incrementing by one gives 123 + 1 = 124.
Thus, the result should be [1,2,4].
"""

from typing import List

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        string1 = ""

        for i in digits:
            string1 += str(i)
        
        string1 = str(int(string1) + 1)
        digits = []
        for i in string1:
            digits.append(int(i))
        return digits
    
sol = Solution()
print(sol.plusOne([1,2,3]))

File: Leetcode/Q66.cpp
/*
You are given a large integer represented as an integer array digits, where each digits[i] is the ith digit of the integer. The digits are ordered from most significant to least significant in left-to-right order. The large integer does not contain any leading 0's.

Increment the large integer by one and return the resulting array of digits.


Example 1:
Input: digits = [1,2,3]
Output: [1,2,4]
Explanation: The array represents the integer 123.
Incrementing by one gives 123 + 1 = 124.
Thus, the result should be [1,2,4].
 */

#include <iostream>
#include <vector>
#include <string.h>

using namespace std;

class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int n = digits.size();
        for(int i=n-1; i>=0; i--) {
            if(digits[i] == 9) {
                digits[i] = 0;
            } else {
                digits[i]++;
                return digits;
            }
        }
        digits.insert(digits.begin(), 1);
        return digits;
    }
};

int main() {

    vector<int> digits = {1,2,9};
    Solution s;
    vector<int> result = s.plusOne(digits);

    for (int digit : result) {
        cout << digit << " ";
    }

    return 0;
}